src/main/java/org/broadinstitute/hellbender/utils/nio/ReadsIterable.java

```diff
@@ -0,0 +1,141 @@
+package org.broadinstitute.hellbender.utils.nio;
+
+import com.google.cloud.storage.contrib.nio.CloudStorageFileSystem;
+import htsjdk.samtools.QueryInterval;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMRecordIterator;
+import htsjdk.samtools.SamInputResource;
+import htsjdk.samtools.SamReader;
+import htsjdk.samtools.SamReaderFactory;
+import htsjdk.samtools.ValidationStringency;
+import htsjdk.samtools.seekablestream.ByteArraySeekableStream;
+import htsjdk.samtools.seekablestream.SeekableStream;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.nio.ChannelAsSeekableStream;
+import org.broadinstitute.hellbender.utils.nio.SeekableByteChannelPrefetcher;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+/**
+ * ReadsIterable gives you all the reads for a given genomic interval.
+ *
+ * QueryInterval + header --> iterable SAMRecords
+ */
+public class ReadsIterable implements Iterable<SAMRecord>, Serializable {
+
+    private static final long serialVersionUID = 1L;
+    final String path;
+    final byte[] index;
+    final QueryInterval interval;
+    final boolean removeHeader = true;
+
+    class ReadsIterator implements Iterator<SAMRecord> {
```

Comment:
We've had mysterious serialization issues with non-static inner classes.  Would you mind making this static?